src/arrays/programs_involving_multidimensional_data/hard/q4/TicTac.java

```diff
@@ -0,0 +1,42 @@
+package arrays.programs_involving_multidimensional_data.hard.q4;
+
+/*
+In the game of Tic-Tac-Toe is played on a board of size 3 by 3 between two players. Each player has a shape that can be
+used (either a circle 'O' or a cross 'X'), and the player has to get three of the same shapes in a row, column, or a
+diagonal to win. Write a Java program that uses a 2D character array to represent the board and randomly assigns one of
+the shapes to each square. For that, write a method called generateTicTac that
```

Comment:
You don't ever specify what the passed integer is used for in the method.

Reply:
You need to add something along the lines of "Generate an n by n Tic-Tac-Toe board ... "